### PR TITLE
find_by_key finder added

### PR DESCRIPTION
**Pull request by tchukuchuk**
Hi Francis,

I have added a find_by_key finder from your example with some specs.

Regards,

Julien

### CODE REVIEW

File: spec/finder_spec.rb [hunk 1]
```diff
@@ -126,13 +126,32 @@ def stub_user(name, age, created_at)
...
+    age = 32
+    created_at = Time.now
+    user = stub_user("Julien", 32, Time.now)
+    user.save
+    user2 = User.find_by_key(user.key)
+    user2.should.not.be.nil
+    user2.key.should == user.key
+    user2.name.should == user.name
+    user2.age.should == user.age
+    #user2.created_at.should == created_at
```
Comment (siuying) [inline spec/finder_spec.rb]: can you remove these two lines ?

**Comment by siuying** [comment]
@tchukuchuk thanks, looks good. 

its okay to remove those extra spaces.

File: spec/finder_spec.rb (hunk 1 above)
Comment (tchukuchuk) [inline spec/finder_spec.rb]: I'm sorry I have just read your comment right now. Which are the **two** lines ?

Comment (siuying) [inline spec/finder_spec.rb]: those commented lines, if they are not part of the test

**Pull request merged**